Remove docs dir

underrun opened the pull request:
outdated docs removal

alexshtin reviewed (approved):
I assume you persisted the info somewhere else.

underrun commented:
@cullywakelin assured me he backed up the docs in case we need them - but this is also git :-) 

@alexshtin I also updated the README to remove a ref to docs/ and adjusted some of the wording if you'd like to take another look

cullywakelin commented on README.md:
```diff
@@ -32,7 +32,7 @@ Try out [Temporal Web UI](https://github.com/temporalio/web) to view your workfl
 We'd love your help in making Temporal great. Please review our [contribution guide](CONTRIBUTING.md).
 
-If you'd like to propose a new feature, first join the Temporal [Slack channel](https://join.slack.com/t/temporalio/shared_invite/zt-c1e99p8g-beF7~ZZW2HP6gGStXD8Nuw) to start a discussion and check if there are existing design discussions. Also peruse our [design docs](docs/design/index.md) in case a feature has been designed but not yet implemented. Once you're sure the proposal is not covered e ...
+If you'd like to propose a new feature, first peruse our [proposals repo](https://github.com/temporalio/proposals) to discover existing active and accepted proposals. Next, join the Temporal [Slack channel](https://join.slack.com/t/temporalio/shared_invite/zt-c1e99p8g-beF7~ZZW2HP6gGStXD8Nuw) to start a discussion or check if feature has already been discussed. Once you're sure the proposal is not ...
```
Maybe also add that new feature requests can be submitted here: https://community.temporal.io/c/feature-requests/6

cullywakelin reviewed (approved):
1 NIT - otherwise, awesome
